frontend/src/pages/hardwareProfiles/manage/validationUtils.ts

```diff
@@ -0,0 +1,37 @@
+import { z } from 'zod';
+import { IdentifierResourceType, TolerationEffect, TolerationOperator } from '~/types';
+
+const identifierSchema = z.object({
+  displayName: z.string(),
+  identifier: z.string(),
+  resourceType: z.nativeEnum(IdentifierResourceType),
```

Comment:
@dpanshug resourceType is `optional` here; this will eventually solve the `create hardware profile` button  been disabled when resourceType is `Other`